### ISSUE TITLE
False positive in Resource Leak Checker when wrapper and wrapped types don't have the same MustCall type

### ISSUE THREAD

**Issue by Nargeshdb** [typetools/checker-framework#4947]
The following example shows a false positive from the Resource Leak Checker.

```java
import org.checkerframework.checker.calledmethods.qual.EnsuresCalledMethods;
import org.checkerframework.checker.mustcall.qual.Owning;
import org.checkerframework.checker.mustcall.qual.MustCall;
import org.checkerframework.checker.mustcall.qual.MustCallAlias;

class MustCallAliasReport {

  @MustCall("a") static class Foo {
    void a() {}
  }

  @MustCall("b") static class FooField {
    private final @Owning Foo finalOwningFoo;

    public @MustCallAlias  FooField(@MustCallAlias Foo f) {
      this.finalOwningFoo = f;
    }

    @EnsuresCalledMethods(
        value = {"this.finalOwningFoo"},
        methods = {"a"})
    void b() {
      this.finalOwningFoo.a();
    }
  }

  void testField() {
    Foo f = new Foo();    // False Positive for this line
    FooField fooField = new FooField(f);
    fooField.b();
  }
}
```

This test passed if I change method `b()` to `a()` in class `FooField` and update its MustCall annotation to `@MustCall("a")`. Thus, I think this problem is because wrapper and wrapped types have different MustCall sets.

**Comment by msridhar** [comment]
@Nargeshdb @kelloggm I remember we discussed this and fixing was more tricky than we anticipated, but I don't remember why.  Do you remember?

**Comment by msridhar** [comment]
Note: the reason why fixing this is not so easy is that `@MustCallAlias` is implemented by relying on built-in support for `@PolyMustCall`, which ends up forcing some `@MustCall` method names to be the same.
